### ISSUE TITLE
Subscribers page freezes with search on site with 4k subscribers

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#88928]
### Quick summary

On a customer's AT site with 4.1k subscribers, Calypso freezes from errors the moment any character is typed into the search field. 

### Steps to reproduce

1. SU into the site: blog_id `75935842`
2. Navigate to https://wordpress.com/subscribers/
3. Type any character into the search field

### What you expected to happen

The search results to appear, paginated 

### What actually happened

The page freezes pretty much immediately: 

![CleanShot 2024-03-26 at 17 04 13@2x](https://github.com/Automattic/wp-calypso/assets/27249804/b8faae35-2592-490b-9068-509f884104f8)

### Impact

One

### Available workarounds?

Yes, easy to implement (download CSV of subscribers instead)

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

Console logs show repeating errors: `DOMException: The operation is insecure.` and `Too many calls to Location or History APIs within a short timeframe.`

Customer tested in multiple browsers, and I can replicate (on their site).

---

Customer report: 7948713-zen

**Comment by cuemarie** [comment]
In a follow up ( 7948796-zen ), I've started the process of testing for plugin conflicts as well, since the site is plugin-enabled.  Did not get a chance to do this in chat.

**Comment by yansern** [comment]
Duplicate of https://github.com/Automattic/wp-calypso/issues/88938.

My suspicion is that there is some Calypso builds (within the past 12-16 hours) range that had a broken react (or react-related dependencies) that causes react to think the searchTerm has been updated when it hasn't, causing a recursive loop.

I was initially able to replicate it, but a couple of hours later and a rebase, this issue was not replicable anymore.

But I've spotted a couple of areas to prevent re-rendering of the action bar / search input, so I made a PR for it https://github.com/Automattic/wp-calypso/pull/88969.
